lib/ImageGallery/ImageGallery.js

```diff
@@ -0,0 +1,279 @@
+    }
+  };
+
+  const isReelScrollable = useMemo(() => {
+    const reel = scrollableReel.current;
+
+    return Boolean(
+      reel &&
+        (reel.scrollWidth > reel.clientWidth ||
+          reel.scrollHeight > reel.clientHeight),
+    );
+  }, [scrollableReel.current]);
+
+  const debouncedOnHoverNavigation = useCallback(
+    debounce(handleThumbnailChange),
+    [],
+  );
+
+  useEffect(() => {
+    const reel = scrollableReel.current;
+    if (reel) {
+      if (isThumbnailsBottom) {
+        const isContainerTotallyScrolled =
+          Math.abs(reel.scrollWidth - reel.clientWidth - reel.scrollLeft) < 25;
+        setScrollForward(!isContainerTotallyScrolled);
+      } else {
+        const isContainerTotallyScrolled =
+          Math.abs(reel.scrollHeight - reel.clientHeight - reel.scrollTop) < 25;
+        setScrollForward(!isContainerTotallyScrolled);
+      }
+    }
+  });
+
+  const moveReelForward = () => {
+    if (scrollableReel.current) {
+      if (isThumbnailsBottom) {
+        scrollableReel.current.scrollLeft += 50;
+        setDragging(true);
+      } else {
+        scrollableReel.current.scrollTop += 50;
+      }
+    }
+  };
+
+  const moveReelBackWard = () => {
+    if (scrollableReel.current) {
+      if (isThumbnailsBottom) {
+        scrollableReel.current.scrollLeft -= 50;
+      } else {
+        scrollableReel.current.scrollTop -= 50;
+      }
+    }
+  };
+
+  return (
+    <div
+      className={clsx({
+        flex: isThumbnailsSide,
+      })}
+    >
+      <div
+        className={clsx(
+          'relative cursor-pointer h-full w-full overflow-hidden',
+          classNames.mainImageWrapper,
+          {
+            'lg:order-2': isThumbnailsSide,
+          },
+        )}
+        {...handlers}
+      >
+        {MainImageComponent ? (
+          MainImageComponent(images[galleryIndex].imageUrl)
+        ) : (
+          <img
+            draggable={false}
+            className="object-cover w-full h-full animate-fade"
+            key={images[galleryIndex].imageUrl}
+            src={images[galleryIndex].imageUrl}
+            alt={images[galleryIndex].alt}
+          />
+        )}
+        {displayArrows && (
+          <>
+            <button
+              className="absolute hidden -translate-y-1/2 md:block left-4 top-1/2"
+              onClick={goBack}
+            >
+              <LeftChevron className="stroke-slate-800" />
+            </button>
+            <button
+              className="absolute hidden -translate-y-1/2 md:block right-4 top-1/2"
+              onClick={goNext}
+            >
+              <RightChevron className="stroke-slate-800" />
+            </button>
+          </>
+        )}
+        {isBestSeller && (
+          <span
+            className={clsx(
+              'absolute px-2 py-1 text-base font-bold text-white uppercase bg-gray-400 top-2 left-2',
+              classNames.bestSellerTag,
+            )}
+          >
+            best seller
```

Comment:
Let's support a custom label here. Instead of an `isBestSeller` boolean prop and a hard coded `best seller` text, how about we accept a `Badge` component prop which, if it's passed, will get rendered here?

This way, developers can pass not only whatever text they want, but also whatever other elements they might want to display inside the badge (maybe icons or dynamic content).